api/types.go

```diff
@@ -731,3 +731,87 @@ func FormatParams(params map[string][]string) (map[string]interface{}, error) {
 
 	return out, nil
 }
+
+type SystemModelInfo struct {
+	// Store is the location where the server stores models in the filesystem.
+	Store string `json:"store"`
+
+	// Count is the number of models currently stored in the filesystem.
+	Count int `json:"count"`
+
+	// FilesystemUsed is the number of bytes used on the filesystem to store models
+	FilesystemUsed uint64 `json:"filesystem_used"`
+
+	// Running is the number of models currently running in the server.
+	Running int `json:"running"`
+
+	// VRAMUsed is the amount of GPU VRAM consumed by the loaded models
+	VRAMUsed uint64 `json:"vram_used"`
+}
+
+type SystemComputeInfo struct {
+	// Cores is the number of detected CPU Cores in the system
+	CPUCores int `json:"cpu_cores"`
+
+	// TotalMemory is the amount of system memory discovered by the server
+	TotalMemory uint64 `json:"total_memory"`
+
+	// FreeMemory is the amount of system memory available for loading new models
+	FreeMemory uint64 `json:"free_memory"`
+
+	// FreeSwap is the amount of swap space available for loading new models
+	FreeSwap uint64 `json:"free_swap"`
+}
+
+type GPUInfo struct {
+	// ID is the unique identifier to use for selection of this specific GPU by device vendor
```

Comment:
With the current discovery code, this is never PCI ID, and in all but 1 case, it's not a sequence number.

For NVIDIA across all platforms it's the device GUID:
- https://github.com/ollama/ollama/blob/main/discover/gpu_info_cudart.c#L134
- https://github.com/ollama/ollama/blob/main/discover/gpu_info_nvcuda.c#L148

For AMD on Linux it's the device GUID:
- https://github.com/ollama/ollama/blob/main/discover/amd_linux.go#L270

Windows AMD is the one exception since I haven't been able to find an API to extract the GUID.  I'm hopeful that 6.2 might improve the situation, but the HIP library on windows seems to be lacking in multiple areas.
- https://github.com/ollama/ollama/blob/main/discover/amd_windows.go#L113